installer/win/WixSharpInstaller/ProgressDialog.cs

```diff
@@ -0,0 +1,218 @@
+        /// (e.g. localize the view).
+        /// </summary>
+        protected override void OnShellChanged()
+        {
+            if (Runtime.Session.IsUninstalling())
+            {
+                dialogText.Text = "[ProgressDlgTitleRemoving]";
+                description.Text = "[ProgressDlgTextRemoving]";
+            }
+            else if (Runtime.Session.IsRepairing())
+            {
+                dialogText.Text = "[ProgressDlgTextRepairing]";
+                description.Text = "[ProgressDlgTitleRepairing]";
+            }
+            else if (Runtime.Session.IsInstalling())
+            {
+                dialogText.Text = "[ProgressDlgTitleInstalling]";
+                description.Text = "[ProgressDlgTextInstalling]";
+            }
+
+            this.Localize();
+        }
+
+        /// <summary>
+        /// Processes the message.
+        /// </summary>
+        /// <param name="messageType">Type of the message.</param>
+        /// <param name="messageRecord">The message record.</param>
+        /// <param name="buttons">The buttons.</param>
+        /// <param name="icon">The icon.</param>
+        /// <param name="defaultButton">The default button.</param>
+        /// <returns></returns>
+        public override MessageResult ProcessMessage(InstallMessage messageType, Record messageRecord, MessageButtons buttons, MessageIcon icon, MessageDefaultButton defaultButton)
+        {
+            switch (messageType)
+            {
+                case InstallMessage.InstallStart:
+                case InstallMessage.InstallEnd:
+                    {
+                        waitPrompt.Visible = false;
+                    }
+                    break;
+
+                case InstallMessage.ActionStart:
+                    {
+                        try
+                        {
+                            //messageRecord[0] - is reserved for FormatString value
+
+                            string message = null;
+
+                            bool simple = true;
+                            if (simple)
+                            {
+                                /*
+                                messageRecord[2] unconditionally contains the string to display
+
+                                Examples:
+
+                                   messageRecord[0]    "Action 23:14:50: [1]. [2]"
+                                   messageRecord[1]    "InstallFiles"
+                                   messageRecord[2]    "Copying new files"
+                                   messageRecord[3]    "File: [1],  Directory: [9],  Size: [6]"
+
+                                   messageRecord[0]    "Action 23:15:21: [1]. [2]"
+                                   messageRecord[1]    "RegisterUser"
+                                   messageRecord[2]    "Registering user"
+                                   messageRecord[3]    "[1]"
+
+                                */
```

Comment:
I think we should remove this commented out code.

Reply:
Well, it is not really commented out code - this file is a modified copy of a file from the Wix# source code, and these lines are documentation/examples, which might be useful for future work, so I think better to leave them in.